modules/build/src/main/scala/scala/build/Inputs.scala

```diff
@@ -243,18 +267,12 @@ object Inputs {
 
     assert(validElems.nonEmpty)
 
-    val (workspace, needsHash) = validElems
-      .collectFirst {
-        case d: Directory => (d.path, true)
-      }
```

Comment:
I would have kept this section. It's not really the problem here.

Reply:
How about this case?
```
$ scala-cli a/ b/ # writes under ./a/.scala-build
$ scala-cli b/ a/  # writes under ./b/.scale-build
```

I think, it should generate only one target. Therefore, I compute common root also for directories.

Reply:
Both look fine to me. The order matters.

I'd expect users to pass the "main" directory first, and we put `.scala-build` in that one.